### ISSUE TITLE
dart:indexed_db from service worker & IdbFactory.supported

### ISSUE THREAD

**Issue by jimmyff** [dart-lang/sdk#31401]
`dart:indexed_db` from service worker IdbFactory.supported doesn't work as `window` is not available from a service worker.

**Comment by vsmenon** [comment]
I don't believe `dart:html, dart:indexed_db` will work in a service worker context without some surgery.

Some folks have worked around using:

https://github.com/isoos/service_worker

@terrylucas

**Comment by jimmyff** [comment]
It's working for me (although I am only using quite a simple implementation). it was only the `IdbFactory.supported` that I ran in to that didn't work.
